parsec/test_utils/organization.py

```diff
@@ -2,70 +2,56 @@
 
 
 import os
-import itertools
-
 import trio
-import pendulum
+from typing import Tuple
+from pathlib import Path
 
-from parsec.crypto import SigningKey
 from parsec.logging import configure_logging
 from parsec.core import logged_core_factory
 from parsec.api.data import UserProfile
-from parsec.api.protocol import DeviceID
-from parsec.core.types import WorkspaceRole, BackendOrganizationBootstrapAddr
-from parsec.core.config import get_default_config_dir, load_config
+from parsec.api.protocol import OrganizationID, HumanHandle, InvitationType
+from parsec.core.types import (
+    WorkspaceRole,
+    BackendAddr,
+    BackendOrganizationBootstrapAddr,
+    BackendInvitationAddr,
+    LocalDevice,
+)
+from parsec.core.config import load_config
 from parsec.core.backend_connection import (
     apiv1_backend_administration_cmds_factory,
     apiv1_backend_anonymous_cmds_factory,
+    backend_authenticated_cmds_factory,
+    backend_invited_cmds_factory,
 )
-from parsec.core.local_device import generate_new_device, save_device_with_password
-from parsec.core.invite_claim import (
-    generate_invitation_token,
-    invite_and_create_device,
-    invite_and_create_user,
-    claim_device,
-    claim_user,
-    InviteClaimError,
+from parsec.core.local_device import save_device_with_password
+from parsec.core.invite import (
+    bootstrap_organization,
+    DeviceGreetInitialCtx,
+    UserGreetInitialCtx,
+    claimer_retrieve_info,
 )
-from parsec.api.data import UserCertificateContent, DeviceCertificateContent
-
-
-async def retry_claim(corofn, *args, retries=10, tick=0.1):
-    for i in itertools.count():
-        try:
-            return await corofn(*args)
-        except InviteClaimError:
-            if i >= retries:
-                raise
-            await trio.sleep(tick)
 
 
 async def initialize_test_organization(
-    backend_address,
-    organization_id,
-    alice_device_id,
-    bob_device_id,
-    other_device_name,
-    alice_workspace,
-    bob_workspace,
-    password,
-    administration_token,
-    force,
-):
+    config_dir: Path,
+    backend_address: BackendAddr,
+    password: str,
+    administration_token: str,
+    force: bool,
+) -> Tuple[LocalDevice, LocalDevice, LocalDevice]:
```

Comment:
We lost quite a bit of parametrization here, but I suspect it wasn't so useful in the first place.

Reply:
yep, the existing params weren't really used.
Besides there provide a kind of "fake genericity" given you can choose the name of the devices, by not their number or their relationship...
So I felt like YAGNI here ;-)